src/ligero/data_structures.rs

```diff
@@ -0,0 +1,427 @@
+        transcript: &mut IOPTranscript<F>,
+    ) -> Result<LigeroPCProofSingle<F, C>, Error> {
+        let t = calculate_t::<F>(sec_param, rho_inv, ext_mat.n)?;
+
+        // 1. left-multiply the matrix by `b`, where for a requested query point `z`,
+        // `b = [1, z^m, z^(2m), ..., z^((m-1)m)]`
+        let v = mat.row_mul(b);
+
+        transcript
+            .append_serializable_element(b"v", &v)
+            .map_err(|_| Error::TranscriptError)?;
+
+        // 2. Generate t column indices to test the linear combination on
+        let indices = get_indices_from_transcript(ext_mat.m, t, transcript)?;
+
+        // 3. Compute Merkle tree paths for the requested columns
+        let mut queried_columns = Vec::with_capacity(t);
+        let mut paths = Vec::with_capacity(t);
+
+        let ext_mat_cols = ext_mat.cols();
+
+        for i in indices {
+            queried_columns.push(ext_mat_cols[i].clone());
+            paths.push(
+                col_tree
+                    .generate_proof(i)
+                    .map_err(|_| Error::TranscriptError)?,
+            );
+        }
+
+        Ok(LigeroPCProofSingle {
+            paths,
+            v,
+            columns: queried_columns,
+        })
+    }
+}
+
+/// The public parameters for the Ligero polynomial commitment scheme.
+/// This is only a default setup with reasonable parameters.
+/// To create your own public parameters, use:
+/// # Example
+/// ```rust
+/// use ark_bls12_377::Fr;
+/// use ark_crypto_primitives::{
+///     crh::{pedersen, sha256::Sha256, CRHScheme, TwoToOneCRHScheme},
+///     merkle_tree::{ByteDigestConverter, Config},
+/// };
+/// use ark_std::test_rng;
+/// use ark_poly_commit::ligero::LigeroPCUniversalParams;
+/// use core::marker::PhantomData;
+///
+/// type LeafH = Sha256;
+/// type CompressH = Sha256;
+/// struct MerkleTreeParams;
+/// impl Config for MerkleTreeParams {
+///     type Leaf = [u8];
+///     type LeafDigest = <LeafH as CRHScheme>::Output;
+///     type LeafInnerDigestConverter = ByteDigestConverter<Self::LeafDigest>;
+///     type InnerDigest = <CompressH as TwoToOneCRHScheme>::Output;
+///     type LeafHash = LeafH;
+///     type TwoToOneHash = CompressH;
+/// }
+/// type MTConfig = MerkleTreeParams;
+/// let mut rng = &mut test_rng();
+/// let leaf_hash_params = <LeafH as CRHScheme>::setup(&mut rng).unwrap();
+/// let two_to_one_params = <CompressH as TwoToOneCRHScheme>::setup(&mut rng)
+///     .unwrap()
+///     .clone();
+/// let pp: LigeroPCUniversalParams<Fr, MTConfig> = LigeroPCUniversalParams::new(128, 2, true,
+///     leaf_hash_params, two_to_one_params);
+#[derive(Derivative, CanonicalSerialize, CanonicalDeserialize)]
+#[derivative(Clone(bound = ""), Debug(bound = ""))]
+pub struct LigeroPCUniversalParams<F: PrimeField, C: Config>
+where
+    C: Config,
+    <<C as Config>::TwoToOneHash as TwoToOneCRHScheme>::Parameters: Debug,
+    <<C as Config>::LeafHash as CRHScheme>::Parameters: Debug,
```

Comment:
Perhaps we should just make `Parameters: Debug` in `crypto-primitives` directly? E.g. a companion to this PR: https://github.com/arkworks-rs/crypto-primitives/pull/114